src/rsa.rs

```diff
@@ -244,6 +255,49 @@ impl RSAKeyPair {
     pub fn public_modulus_len(&self) -> usize {
         unsafe { RSA_size(self.rsa.as_ref()) }
     }
+
+    /// Sign the given message, hashed with the given digest algorithm.
+    pub fn sign(&self, digest_alg: &'static digest::Algorithm, msg: &[u8], signature: &mut [u8]) -> Result<(), ()> {
+        if signature.len() != self.public_modulus_len() {
+            return Err(());
+        }
+
+        let found = PREFIXES.into_iter().find(|pair| pair.0.nid == digest_alg.nid);
+        if found.is_none() {
+            return Err(());
+        }
+
+        let msg_len = self.public_modulus_len();
+        let hashed = digest::digest(digest_alg, msg);
+        let prefix = found.unwrap().1;
+        let digest_len = prefix.len() + hashed.as_ref().len();
+        if msg_len < digest_len + 11 {
+            return Err(());
+        }
+
+        let padding_len = msg_len - digest_len - 3;
+        let mut pad_bytes: Vec<u8> = Vec::with_capacity(msg_len);
+        pad_bytes.push(0);
+        pad_bytes.push(1);
+        for _ in 0..padding_len {
+            pad_bytes.push(255);
+        }
+        pad_bytes.push(0);
+        pad_bytes.extend(prefix);
+        pad_bytes.extend(hashed.as_ref());
+
+        let rng = rand::SystemRandom::new();
```

Comment:
`sign` should take a `rng: &SecureRandom` parameter, instead. In general, we never use `SystemRandom::new()`, but require the library user to do that.